loky/backend/utils.py

```diff
@@ -84,3 +94,207 @@ def _recursive_terminate(pid):
             # level function raise a warning and retry to kill the process.
             if e.errno != errno.ESRCH:
                 raise
+
+
+##############################################################################
+# The following code is derived from code by Intel developper @anton-malakhov
+# available at https://github.com/IntelPython/smp
+#
+# Copyright (c) 2017, Intel Corporation published under the BSD 3-Clause
+# license
+#
+
+
+# Structure to cast the the info on dynamically loaded library.
+if sys.maxsize > 2**32:
+    class dl_phdr_info(ctypes.Structure):
+        _fields_ = [("dlpi_addr",  ctypes.c_uint64),
+                    ("dlpi_name",  ctypes.c_char_p),
+                    ("dlpi_phdr",  ctypes.c_void_p),
+                    ("dlpi_phnum", ctypes.c_uint16)]
+else:
+    class dl_phdr_info(ctypes.Structure):
+        _fields_ = [("dlpi_addr",  ctypes.c_uint32),
+                    ("dlpi_name",  ctypes.c_char_p),
+                    ("dlpi_phdr",  ctypes.c_void_p),
+                    ("dlpi_phnum", ctypes.c_uint16)]
+
+
+# This function is called on each dynamically loaded library attached to our
+# process. It checks if any of this library matches the name given in data and
+# if a match is found, it set the global variable _MODULE_NAME to the full path
+# of this library, to be loaded with ctypes.CDLL.
+def check_module(info, size, data):
```

Comment:
This should be renamed to `match_module_callback`.

Reply:
Also, I think the variable names in this function are very confusing. 

- `data` and `desired_module` should be renamed to `module_name`,
- `module_name` should be renamed to `module_path`.
- `_MODULE_NAME` should be renamed to `_MODULE_PATH` (and should be made a thread local).